pandas/core/dtypes/cast.py

```diff
@@ -1747,3 +1747,32 @@ def validate_numeric_casting(dtype: np.dtype, value):
     ):
         if is_bool(value):
             raise ValueError("Cannot assign bool to float/integer series")
+
+
+def safe_convert_to_ndarray(values) -> np.ndarray:
```

Comment:
To type this, I think `AnyArrayLike` would work, as it's defined as
```
AnyArrayLike = TypeVar("AnyArrayLike", "ExtensionArray", "Index", "Series", np.ndarray)
```

Reply:
I ended up dropping support for `Series` like discussed above and thus ended up with `Union[ArrayLike, Index]`.